sapmon/payload/provider/saphana.py

```diff
@@ -68,8 +69,11 @@ def parseProperties(self):
          try:
             vaultNameSearch = re.search("https://(.*).vault.azure.net", hanaDbPasswordKeyVaultUrl)
             kvName = vaultNameSearch.group(1)
+            passwordSearch = re.search("https://.*.vault.azure.net/secrets/(.*)/(.*)", hanaDbPasswordKeyVaultUrl)
+            passwordName = passwordSearch.group(1)
+            passwordVersion = passwordSearch.group(2)
```

Comment:
- I would group those two searches together
- Dots inside the urls should be properly escaped
- re.match is better than re.search here: https://docs.python.org/3/library/re.html#search-vs-match
- adding re.IGNORECASE

```suggestion
            vaultUrlMatch = re.match("https://(.*)\.vault\.azure\.net/secrets/(.*)/(.*)", hanaDbPasswordKeyVaultUrl, re.IGNORECASE)
            kvName = vaultUrlMatch.group(1)
            passwordName = vaultUrlMatch.group(2)
            passwordVersion = vaultUrlMatch.group(3)
```

Reply:
Good comments. I've updated the regex and made version optional.